### PR TITLE
Update meteor pipelines

### LINKED ISSUE #90: Tech debt/backporting tracker
Umbrella issue for tech debt and backporting:

- [x] Remove deployment logic from the operator and move it to a tekton pipeline
- [ ] Backport `.aicoe.yaml` analysis to notebook pipeline in AICoE
- [ ] Backport restructuring changes to AICoE pipelines from meteor

### PR DESCRIPTION
**Pull request by tumido**
Resolves https://github.com/AICoE/meteor/issues/90

1. Pass new params to pipelines
2. Remove external namespace role management since helm charts and releases doesn't work well cross namespace - logic moved to the operator
3. Backport https://github.com/AICoE/meteor-operator/pull/47
4. Backport push gateway
5. Add pipeline descriptions/annotations in support of https://github.com/AICoE/meteor-operator/issues/52

### CODE REVIEW

**Comment by tumido** [comment]
I have no idea why the lint is failing...

image

**Comment by codificat** [comment]
Thanks for this! Looking...
/assign

**Comment by goern** [comment]
@tumido do you think we should refactor all the scripts (bash or python) into scripts and put them in a container image, so that the tekton yaml looks cleaner. or is that something that should be done before a release, as it is slowing development down a little?

**Comment by tumido** [comment]
I think we can/should do it rather later, it's better to prototype with tasks than depend on image changes + image release + task adjustment. Now it's just a task adjustment.. WDYT? :slightly_smiling_face:

File: charts/meteor-pipelines/templates/meteor_buildah.yaml
Comment (codificat) [inline charts/meteor-pipelines/templates/meteor_buildah.yaml]: Why is that? Shouldn't the default be "true" here?

**Review by codificat — CHANGES_REQUESTED** [review]
Only one comment/question about the default TLS setting

**Comment by codificat** [comment]
> I have no idea why the lint is failing...

it seems to be not configured. #11 should fix that

**Comment by codificat** [comment]
@tumido the PR checks should look better here if you rebase now that #10 and #11 are in

File: charts/meteor-pipelines/templates/meteor_buildah.yaml
Comment (tumido) [reply charts/meteor-pipelines/templates/meteor_buildah.yaml]: Yeah.. I was wondering the same.. I had to set this to false it because the builds started failing in TLS. Checking on the rest of the build tasks in this repo - all have TLS set to false by default (and never changed to true):

https://github.com/thoth-station/helm-charts/blob/30ab8051d7e75d178e4e1b77acfaf1663586729c/charts/meteor-pipelines/templates/issue-release-task.yaml#L11-L15

https://github.com/thoth-station/helm-charts/blob/30ab8051d7e75d178e4e1b77acfaf1663586729c/charts/meteor-pipelines/templates/overlay-build-task.yaml#L50-L54

https://github.com/thoth-station/helm-charts/blob/30ab8051d7e75d178e4e1b77acfaf1663586729c/charts/meteor-pipelines/templates/pr-build-release.yaml#L47-L51

https://github.com/thoth-station/helm-charts/blob/30ab8051d7e75d178e4e1b77acfaf1663586729c/charts/meteor-pipelines/templates/pr-build.yaml#L31-L35

https://github.com/thoth-station/helm-charts/blob/30ab8051d7e75d178e4e1b77acfaf1663586729c/charts/meteor-pipelines/templates/tag-build.yaml#L8-L12

https://github.com/thoth-station/helm-charts/blob/30ab8051d7e75d178e4e1b77acfaf1663586729c/charts/meteor-pipelines/templates/tag-release-task.yaml#L44-L48

maybe this is a part of a bigger underlying issue and we should pursue TLS_VERIFY=true always.. I'll file an extra issue for the other pipelines and do some tests for the meteor one - if it can be enabled, I'll just enable it :slightly_smiling_face:

/hold

**Comment by tumido** [comment]
@codificat rebased and still failing because the chart version is not changed.. I don't think we want to change the chart version with every PR. We want to do that on releases though.. wdyt?

**Comment by tumido** [comment]
Oh.. I take that back.. the way it's set up is that we release on very push to master.. in that case I need to bump the version :slightly_smiling_face:

**Comment by tumido** [comment]
CI failure unrelated to the changeset.

**Comment by codificat** [comment]
The linter is complaining about "too few spaces before comment" (https://github.com/thoth-station/helm-charts/runs/4383627325?check_suite_focus=true#step:7:24). 

This is in `charts/meteor-pipelines/values.yaml`:
https://github.com/thoth-station/helm-charts/blob/2e536885dfeed90d37e664c00c212540c67971c0/charts/meteor-pipelines/values.yaml#L16

Adding an additional space between `:` and `#` should make the linter happy.

I know this is not part of your changeset, but if you can add that one char to your PR I believe it would be simpler than a separate PR + rebase.

**Comment by tumido** [comment]
Ci failing again:
image

I don't think this is caused by the PR...

**Comment by codificat** [comment]
> I don't think this is caused by the PR...

You are right, chart testing apparently has never worked for this repo yet.

To fix this we can either disable maintainer validation in the chart test config, or update the metadata to use github usernames so that they can be vaildated (background [here](https://github.com/helm/chart-testing/issues/192#issuecomment-661901643).

Assuming the latter, the suggestion is to use `goern` as the maintainer name in the Chart.

**Comment by tumido** [comment]
Thanks @codificat I've merged your PR. That made the lint to pass. Now the CI is failing on e2e tests (seems like Tekton is not installed):
image

Can we maybe debug it in a separate PR and just bypass it for this one? While I greatly appreciate CI and checks, I think that right now, when it's not yet properly set up, it's very disruptive and delaying any work... :confused:

**Comment by codificat** [comment]
Agreed, let's get that in and address CI separately, thanks for the patience :sweat_smile: 

/approve
/override lint-test

**Comment by codificat** [comment]
/unhold
as we will work on the remaining issues later

**Pull request merged**
